priv/tools/internal/block_audit.erl

```diff
@@ -0,0 +1,236 @@
+%% ---------------------------------------------------------------------
+%%
+%% Copyright (c) 2007-2015 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% ---------------------------------------------------------------------
+
+-module(block_audit).
+
+-mode(compile).
+
+-export([main/1]).
+-export([info/2, verbose/3]).
+
+-define(SLK_TIMEOUT, 360000000). %% 100 hours
+
+-include_lib("riak_cs/include/riak_cs.hrl").
+
+-record(buuid, {uuid  :: binary(),
+                seqs  :: [non_neg_integer()] % sequence numbers
+               }).
+
+main(Args) ->
+    _ = application:load(lager),
+    ok = application:set_env(lager, handlers, [{lager_console_backend, info}]),
+    ok = lager:start(),
+    {ok, {Options, _PlainArgs}} = getopt:parse(option_spec(), Args),
+    LogLevel = case proplists:get_value(debug, Options) of
+                   0 ->
+                       info;
+                   _ ->
+                       ok = lager:set_loglevel(lager_console_backend, debug),
+                       debug
+               end,
+    debug("Log level is set to ~p", [LogLevel]),
+    debug("Options: ~p", [Options]),
+    case proplists:get_value(host, Options) of
+        undefined ->
+            getopt:usage(option_spec(), "riak-cs escript /path/to/block_audit.erl"),
+            halt(1);
+        Host ->
+            Port = proplists:get_value(port, Options),
+            debug("Connecting to Riak ~s:~B...", [Host, Port]),
+            case riakc_pb_socket:start_link(Host, Port) of
+                {ok, Pid} ->
+                    pong = riakc_pb_socket:ping(Pid),
+                    audit(Pid, Options),
+                    timer:sleep(100);
```

Comment:
It may be old/historical workaround now. R15B01 (too old one :sweat_smile: ) fixed standard out flush issue when escript exits [1].
There is (small) possibility to backport this script to CS 1.5 env, so if you don't mind, I want to
leave this line as is. Thoughts?

[1]  OTP-9985 http://www.erlang.org/download/otp_src_R15B01.readme